src/http/adapter.js

```diff
@@ -68,6 +68,14 @@ function checkPermissionFor( state, user, context, action ) {
 		.then( onPermission, onError );
 }
 
+function executeStack( resource, envelope, stack ) {
+	var calls = stack.slice( 0 );
+	var call = calls.shift();
+	return call ?
+		call.bind( resource )( envelope, executeStack.bind( undefined, resource, envelope, calls ) ) :
+		{ status: 500, data: { message: 'The server failed to provide a response' } };
+}
```

Comment:
Maybe I'm reading it wrong? It looks as though if you did `next(); return { status:404 };` (ignoring the guidelines in the readme) that the result would be a 404 but any additional middleware would continue to execute. Might not be an issue or worth addressing, but curious about your thoughts.

For example, in this stack the `sweetestMiddlewareEvar` (which could be writing data, metrics, logs, etc.) would always execute but it's response would not be used:
```js
{
   middleware: [ 
        function(env, next) { next(); return { status:404 }; }, 
        sweetestMiddlewareEvar
    ]
}
```